src/dicom-web/StudyVolumeDicomWeb.vue

```diff
@@ -0,0 +1,237 @@
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  reactive,
+  set,
+  toRefs,
+  watch,
+} from '@vue/composition-api';
+import type { PropType } from '@vue/composition-api';
+import { useDicomMetaStore } from './dicom-meta.store';
+import {
+  useDicomWebStore,
+  isDownloadable,
+  VolumeProgress,
+} from './dicom-web.store';
+
+const percentDone = (progress: VolumeProgress): number => {
+  if (!progress || progress.total === 0) return 0;
+  const { total, loaded } = progress;
+  return Math.round((100 * loaded) / total);
+};
+
+// from https://stackoverflow.com/a/18650828
+function formatBytes(bytes: number, decimals = 2) {
```

Comment:
I'd add this to `utils`, as it can be a generally useful utility function down the road.

Reply:
Moved to utils.